lib/read/lib/Cardano/Wallet/Read/Block/Gen/Byron.hs

```diff
@@ -0,0 +1,142 @@
+import Cardano.Api
+    ( ByronEra
+    )
+import Cardano.Chain.Block
+    ( mkHeaderExplicit
+    )
+import Cardano.Chain.Slotting
+    ( EpochSlots
+    )
+import Cardano.Crypto.Hashing
+    ( serializeCborHash
+    )
+import Cardano.Crypto.ProtocolMagic
+    ( ProtocolMagicId (..)
+    )
+import Cardano.Crypto.Signing
+    ( SigningKey (..)
+    )
+import Cardano.Wallet.Read.Block.BlockNo
+    ( BlockNo (..)
+    )
+import Cardano.Wallet.Read.Block.Gen.BlockParameters
+    ( BlockParameters (..)
+    )
+import Cardano.Wallet.Read.Block.SlotNo
+    ( SlotNo (..)
+    )
+import Cardano.Wallet.Read.Tx
+    ( Tx (..)
+    )
+import Data.Coerce
+    ( coerce
+    )
+import Data.Text
+    ( Text
+    )
+import Test.Cardano.Crypto.CBOR
+    ( getBytes
+    )
+import Test.Cardano.Ledger.Alonzo.AlonzoEraGen
+    ()
+
+import qualified Cardano.Chain.Block as Byron
+import qualified Cardano.Chain.Common as Byron
+import qualified Cardano.Chain.Delegation as Byron
+import qualified Cardano.Chain.Delegation as Byron.Delegation
+import qualified Cardano.Chain.Slotting as L
+import qualified Cardano.Chain.Ssc as Byron
+import qualified Cardano.Chain.Update as Byron
+import qualified Cardano.Chain.Update as Byron.Update
+import qualified Cardano.Chain.UTxO as Byron
+import qualified Cardano.Crypto.Signing as S
+import qualified Cardano.Crypto.Wallet as CC
+import qualified Ouroboros.Consensus.Byron.Ledger as O
+
+mkByronBlock :: BlockParameters ByronEra -> O.ByronBlock
+mkByronBlock
+    BlockParameters
+        { slotNumber = SlotNo slotNo
+        , blockNumber = BlockNo blockNo
+        , txs
+        } =
+        O.annotateByronBlock eslots
+            $ Byron.ABlock (header slotNo' blockNo' body) body mempty
+      where
+        body :: Byron.ABody ()
+        body = Byron.ABody txs' Byron.SscPayload delegation update
+
+        txs' :: Byron.ATxPayload ()
+        txs' = Byron.ATxPayload $ fmap unTx txs
+
+        slotNo' :: L.SlotNumber
+        slotNo' = L.SlotNumber $ fromIntegral slotNo
+
+        blockNo' :: Byron.ChainDifficulty
+        blockNo' = Byron.ChainDifficulty $ fromIntegral blockNo
+
+header :: L.SlotNumber -> Byron.ChainDifficulty -> Byron.Body -> Byron.Header
+header slotNumber blockNumber body =
+    mkHeaderExplicit
+        protocolMagicId
+        headerHash
+        blockNumber
+        eslots
+        slotNumber
+        issuerSk
+        delegationCertificate
+        body
+        protocolVersion
+        softwareVersion
+
+delegation :: Byron.Delegation.APayload ()
+delegation = Byron.Delegation.UnsafeAPayload mempty mempty
+
+update :: Byron.Update.APayload ()
+update = Byron.Update.APayload Nothing mempty mempty
+
+eslots :: EpochSlots
+eslots = L.EpochSlots 20
+
+protocolMagicId :: ProtocolMagicId
+protocolMagicId = ProtocolMagicId 7
+
+headerHash :: Byron.HeaderHash
+headerHash = coerce $ serializeCborHash ("HeaderHash" :: Text)
+
+protocolVersion :: Byron.ProtocolVersion
+protocolVersion = Byron.ProtocolVersion 1 1 1
```

Comment:
```suggestion
protocolVersion :: Byron.ProtocolVersion
protocolVersion = Byron.ProtocolVersion 3 0 0
```

To avoid random breakage, I would align this with the Cardano Mainnet parameters.

The protocol version is specified in the Mainnet node config file.

Here is code for parsing the config file:
```hs
module Cardano.Api.LedgerState where
…

parseByronProtocolVersion :: Object -> Parser Cardano.Chain.Update.ProtocolVersion
parseByronProtocolVersion o =
  Cardano.Chain.Update.ProtocolVersion
    <$> o .: "LastKnownBlockVersion-Major"
    <*> o .: "LastKnownBlockVersion-Minor"
    <*> o .: "LastKnownBlockVersion-Alt"
```

And here is the corresponding section of the mainnet config file

https://github.com/input-output-hk/cardano-node/blob/db0f82960be3b3b1d47e76c9f8ba22c32915f2e7/configuration/cardano/mainnet-config.json#L8-L10